### DISCUSSION TITLE
Defining new parameter names

### DISCUSSION THREAD

**Discussion by danteppc** [bitcraze/discussions#170] [Q&A] [answered]
Hello,

We are trying to define a new parameter in the crazyflie firmware using ADD_PARAM in an app like this:

[
#define DEBUG_MODULE ""

#include <string.h>
#include <stdint.h>
#include <stdbool.h>
#include <stdio.h>
#include "app.h"
#include "FreeRTOS.h"
#include "task.h"
#include "log.h"
#include "param.h"
#include "debug.h"
#include "locodeck.h"
#include "lpsTdoa2Tag.h"

uint8_t myapp_counter = 0;

void appMain() {
  while(1) {
	// code
  }
}

PARAM_GROUP_START(app)
ADD_PARAM(PARAM_UINT8, myappcnt, &myapp_counter)
PARAM_GROUP_STOP(app)
](url)

but we get the following errors:

`  GEN     ./Makefile
scripts/kconfig/conf  --silentoldconfig Kconfig
  Using /Users/mahyar/Documents/GitHub/crazyflie-firmware as source for kernel
  CHK     include/generated/version.h
  GEN     ./Makefile
  CC      /Users/mahyar/Documents/GitHub/crazyflie-firmware/examples/someApp/src/myapp.o
/Users/mahyar/Documents/GitHub/crazyflie-firmware/examples/someApp/src/myapp.c:108:1: error: implicit declaration of function 'ADD_PARAM' [-Werror=implicit-function-declaration]
  108 | ADD_PARAM(PARAM_UINT8, myappcnt, &myapp_counter)
      | ^~~~~~~~~
/Users/mahyar/Documents/GitHub/crazyflie-firmware/examples/someApp/src/myapp.c:108:24: error: 'myappcnt' undeclared here (not in a function)
  108 | ADD_PARAM(PARAM_UINT8, myappcnt, &myapp_counter)
      |                        ^~~~~~~~
In file included from /Users/mahyar/Documents/GitHub/crazyflie-firmware/examples/someApp/src/myapp.c:15:
/Users/mahyar/Documents/GitHub/crazyflie-firmware/src/modules/interface/param.h:131:3: error: expected '}' before '{' token
  131 |   { \
      |   ^
/Users/mahyar/Documents/GitHub/crazyflie-firmware/src/modules/interface/param.h:135:3: note: in expansion of macro 'PARAM_ADD_GROUP'
  135 |   PARAM_ADD_GROUP(PARAM_GROUP | PARAM_STOP, stop_##NAME, 0x0) \
      |   ^~~~~~~~~~~~~~~
/Users/mahyar/Documents/GitHub/crazyflie-firmware/examples/someApp/src/myapp.c:109:1: note: in expansion of macro 'PARAM_GROUP_STOP'
  109 | PARAM_GROUP_STOP(app)
      | ^~~~~~~~~~~~~~~~
/Users/mahyar/Documents/GitHub/crazyflie-firmware/src/modules/interface/param.h:141:93: note: to match this '{'
  141 |   static struct param_s __params_##NAME[] __attribute__((section(".param." #NAME), used)) = { \
      |                                                                                             ^
/Users/mahyar/Documents/GitHub/crazyflie-firmware/examples/someApp/src/myapp.c:107:1: note: in expansion of macro 'PARAM_GROUP_START'
  107 | PARAM_GROUP_START(app)
      | ^~~~~~~~~~~~~~~~~
cc1: error: unrecognized command line option '-Wno-stringop-overread' [-Werror]
cc1: all warnings being treated as errors
`
Any thoughts?

**Comment by danteppc** [comment, marked as answer]
Sorry, we just realized we should be using "PARAM_ADD" not "ADD_PARAM".

**Reply by knmcguire** [reply]
ah great you figured it out! The error message is indeed not super clear
